plat/rockchip/common/aarch64/plat_helpers.S

```diff
@@ -192,20 +195,23 @@ endfunc plat_crash_console_putc
 	.align	16
 func platform_cpu_warmboot
 	mrs	x0, MPIDR_EL1
-	and	x1, x0, #MPIDR_CPU_MASK
-	and	x0, x0, #MPIDR_CLUSTER_MASK
+	and	x19, x0, #MPIDR_CPU_MASK
+	and	x20, x0, #MPIDR_CLUSTER_MASK
+#if NEED_PMU_MACROS_S
+	func_rockchip_clst_warmboot
+#endif
 	/* --------------------------------------------------------------------
 	 * big cluster id is 1
 	 * big cores id is from 0-3, little cores id 4-7
 	 * --------------------------------------------------------------------
 	 */
-	add	x0, x1, x0, lsr #6
+	add	x21, x19, x20, lsr #PLATFORM_CLST_TO_CPUID_SHIFT
```

Comment:
I don't really understand your comment. This instruction doesn't care what the previous value of x21 (or x0) is. But it's only a minor thing so I won't push it.